File: java/src/main/java/com/ellen/java/jmvp/base/BaseModel.java
package com.ellen.java.jmvp.base;

public abstract class BaseModel{
    /**
     * 被Presenter绑定时调用
     */
    protected void attachedByPresenter(){}

    /**
     * 被Presenter解绑时调用
     */
    protected void detachedByPresenter(){}
}

File: app/src/main/java/com/ellen/androidarchitecture/kmvp/base/model/BaseModel.kt
package com.ellen.androidarchitecture.kmvp.base.model

/**
 * MVP中的M层
 */
abstract class BaseModel{


    /**
     * 绑定时调用
     */
    open fun attachedByPresenter(){}

    /**
     * 取消绑定时调用
     */
    open fun detachedByPresenter(){}
}